src/utils/deep_research.py

```
@@ -0,0 +1,377 @@
+from browser_use.agent.service import Agent
+from browser_use.browser.browser import BrowserConfig, Browser
+from browser_use.agent.views import ActionResult
+from browser_use.browser.context import BrowserContext
+from browser_use.controller.service import Controller, DoneAction
+from main_content_extractor import MainContentExtractor
+from langchain.schema import SystemMessage, HumanMessage
+from json_repair import repair_json
+from src.agent.custom_prompts import CustomSystemPrompt, CustomAgentMessagePrompt
+from src.controller.custom_controller import CustomController
+from src.browser.custom_browser import CustomBrowser
+from src.browser.custom_context import BrowserContextConfig, BrowserContext
```

Ruff (0.8.2)

Redefinition of unused `BrowserContext` from line 19

(F811)